nodejs/awsx/autoscaling/autoscaling.ts

```diff
@@ -60,24 +62,37 @@ export class AutoScalingGroup extends pulumi.ComponentResource {
         this.targetGroups = args.targetGroups || [];
         const targetGroupArns = this.targetGroups.map(g => g.targetGroup.arn);
 
+        let launchConfigurationId: pulumi.Input<string>;
+        let stackName: pulumi.Input<string>;
+
         // Use the autoscaling config provided, otherwise just create a default one for this cluster.
-        if (args.launchConfiguration) {
-            this.launchConfiguration = args.launchConfiguration;
+        if (args.launchConfigurationId) {
+            launchConfigurationId = args.launchConfigurationId;
+            stackName = new random.RandomId(name, {
+                byteLength: 8,
+                prefix: name,
+                keepers: { launchConfigurationId },
+            }, { parent: this }).b64Url;
         }
         else {
-            this.launchConfiguration = new AutoScalingLaunchConfiguration(
-                name, this.vpc, args.launchConfigurationArgs, { parent: this });
+            this.launchConfiguration = args.launchConfiguration
+                ? args.launchConfiguration
+                : new AutoScalingLaunchConfiguration(
+                    name, this.vpc, args.launchConfigurationArgs, { parent: this });
+
+            launchConfigurationId = this.launchConfiguration.id;
+            stackName = this.launchConfiguration.stackName;
         }
 
         this.vpc = args.vpc || x.ec2.Vpc.getDefault({ parent: this });
 
         // Use cloudformation to actually construct the autoscaling group.
         this.stack = new aws.cloudformation.Stack(name, {
             ...args,
-            name: this.launchConfiguration.stackName,
+            name: stackName,
             templateBody: getCloudFormationTemplate(
                 name,
-                this.launchConfiguration.id,
+                launchConfigurationId,
```

Comment:
open question if it's sufficient to just provide a stack-name/launch-config-id.  Or if there might be other information necessary to pass in by customers to get the experience they need.